### PR TITLE
Refactors special abilities to use signals

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
Converts this to signals in prep for #1324

Also just generic handling of special abilities of several castes

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 1]
```diff
@@ -194,25 +189,15 @@
...
-					add_slowdown(1.5)
-				M.stealth_router(HANDLE_STEALTH_CODE_CANCEL)
-			M.neuroclaw_router(src) //if we have neuroclaws...
-			if(dam_bonus)
-				tackle_pain += dam_bonus
 
-			tackle_pain = M.hit_and_run_bonus(tackle_pain) //Apply Runner hit and run bonus damage if applicable
+			var/list/modified_damage = list()
+
+			SEND_SIGNAL(M, COMSIG_XENO_HUMAN_SLASH, src, tackle_pain, modified_damage, intent)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm]: Won't this produce an empty `modified_damage` list?

**Comment by Rohesie** [comment]
Set testing label once conflict is resolved.

**Comment by spookydonut** [comment]
waiting on attack_alien pr first

**Comment by DominikPanic** [comment]
How much has this been tested?
A quick thunderdome test revealed that:
stealth attack doesn't work
pounce moves you but only pushes the person while stunning you and not knocking them down

File: code/game/objects/structures/barricade.dm [hunk 2]
```diff
@@ -136,8 +136,7 @@
 		"<span class='danger'>The barbed wire slices into you!</span>", null, 5)
 		M.apply_damage(10)
 	update_health(TRUE)
-	if(M.stealth_router(HANDLE_STEALTH_CHECK)) //Cancel stealth if we have it due to aggro.
-		M.stealth_router(HANDLE_STEALTH_CODE_CANCEL)
+	SEND_SIGNAL(M, COMSIG_XENO_ATTACK_DECLOAK)
```
[the remark this answers is not in the document]
Comment (spookydonut) [reply code/game/objects/structures/barricade.dm]: eventually it'll become a xeno_hostile_action at the base on attack_alien but thats pending other refactoring.

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 3]
```diff
@@ -67,34 +73,21 @@
...
-			KnockOut(knockout_stacks)
-			adjust_stagger(staggerslow_stacks)
-			add_slowdown(staggerslow_stacks)
-		X.stealth_router(HANDLE_STEALTH_CODE_CANCEL)
-	X.neuroclaw_router(src) //if we have neuroclaws...
+
+	var/list/modified_damage = list()
+	var/list/modified_armor = list()
+
+	SEND_SIGNAL(X, COMSIG_XENO_LIVING_SLASH, src, tackle_pain, modified_damage, armor_block, modified_armor, INTENT_DISARM)
```
[the remark this answers is not in the document]
Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/attack_alien.dm]: ok

**Pull request closed without merging**